src/main/java/uk/ac/ic/wlgitbridge/io/http/ning/NingHttpClient.java

```diff
@@ -52,11 +52,16 @@ public State onBodyPartReceived(
                 @Override
                 public byte[] onCompleted(
                         Response response
-                ) throws IOException {
+                ) throws Exception {
```

Comment:
Is there a more specific type of Exception that we could throw here?

Reply:
Good question. Do you mean that I should find an exception provided by Java, or that I should make a custom exception class?

Reply:
I suppose a standard exception would be best. However, if there are no obvious candidates, then I think the code is ok as it is (especially as it is only called from one location)

Reply:
Ok. Thanks. I'll look it up.

Reply:
As far as I can tell, standard exceptions in Java are pretty specific ([java.lang docs](https://docs.oracle.com/javase/8/docs/api/java/lang/package-summary.html)). There isn't a good candidate for this case. I'll leave it as is since we don't really need to do control flow based on the kind of exception that is thrown.